frontend/src/components/Game.js

```diff
@@ -1,13 +1,198 @@
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    fetch(
+      `${BACKEND_URL}/repository/${repositoryId}/tree/${directoryId}`,
+      options
+    )
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        return response.json().then((e) => {
+          throw new Error(e.detail || "Unknown error");
+        });
+      })
+      .then((directory) => {
+        const { subdirectories } = directory;
+        setDirectories(subdirectories);
+        setPathHistory((prevPathHistory) => [
+          ...prevPathHistory,
+          {
+            path: currentPath,
+            directories: directories,
+          },
+        ]);
+        setCurrentPath((prevPath) => `${prevPath}/${directoryName}`);
+      })
+      .catch((e) => {
+        const message = "Error occurred: " + e.message;
+        console.log(message);
+        setError(message);
+      });
+  };
+
+  useEffect(() => {
+    const options = {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    fetch(`${BACKEND_URL}/game/${gameId}`, options)
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        return response.json().then((e) => {
+          throw new Error(e.detail || "Unknown error");
+        });
+      })
+      .then((game) => {
+        const { player_name, repository_id } = game;
+        setPlayerName(player_name);
+        setRepositoryId(repository_id);
+
+        fetch(`${BACKEND_URL}/repository/${repository_id}/tree`, options)
+          .then((response) => {
+            if (response.ok) {
+              return response.json();
+            }
+            return response.json().then((e) => {
+              throw new Error(e.detail || "Unknown error");
+            });
+          })
+          .then((rootDirectory) => {
+            const { subdirectories } = rootDirectory;
+            setDirectories(subdirectories);
+          })
+          .catch((e) => {
+            const message = "Error occurred: " + e.message;
+            console.log(message);
+            setError(message);
+          });
+      })
+      .catch((e) => {
+        const message = "Error occurred: " + e.message;
+        console.log(message);
+        setError(message);
+      });
+  }, [gameId]);
+
+  const handleClickChoose = (directoryName) => {
+    console.log(`Chose directory: ${directoryName}`);
+    setAnswer(directoryName);
+  };
+
+  const handleSubmit = (answer) => {
+    if (answer) {
+      const options = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          answer: answer,
+        }),
+      };
+
+      fetch(`${BACKEND_URL}/game/${gameId}`, options)
+        .then((response) => {
+          if (response.ok) {
+            setAnswerSubmitted(true);
```

Comment:
I don't think we need the `answerSubmitted` flag. I think that after submitting the answer, the player should be automatically redirected to the results.